### PR TITLE
Cleanup GoogleCloudStorage  implementations

### PR DESCRIPTION
**Pull request by medb**
Changes in this PR:
- rename `GcsJavaClientImpl` to `GoogleCloudStorageClientLibraryImpl` for consistency
- migrate `GoogleCloudStorageClientLibraryImpl` to `ForwardingGoogleCloudStorage`
- remove obsolete `GoogleCloudStorageNativeFileSystem` and `GoogleCloudStorageFileSystemProvider`
- replace boolean `fs.gs.javaclient.enable` property with enum `fs.gs.client.type` property

### CODE REVIEW

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientLibraryImpl.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package com.google.cloud.hadoop.gcsio;
```
Comment (arunkumarchacko) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientLibraryImpl.java:1]: please add copyright

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemOptions.java [hunk 2]
```diff
@@ -24,30 +24,29 @@ public abstract class GoogleCloudStorageFileSystemOptions {
...
   public static GoogleCloudStorageFileSystemOptions DEFAULT = builder().build();
 
-  public enum FilesystemAPI {
-    OBJECT,
-    DIRECTORY,
-    AUTO
+  /** Cloud Storage client to use. */
+  public enum ClientType {
+    HTTP_API_CLIENT,
+    CLIENT_LIBRARY
```
Comment (arunkumarchacko) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemOptions.java:30]: do we want to change this to JAVA_STORAGE_CLIENT or STORAGE_CLIENT to match https://github.com/googleapis/java-storage ?

Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemOptions.java]: Done

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientLibraryImpl.java (hunk 1 above)
Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientLibraryImpl.java]: Done

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 Google LLC
```
Comment (arunkumarchacko) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java:2]: nit: 2023? :)

**Review by arunkumarchacko — APPROVED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java (hunk 3 above)
Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java:2]: This file was renamed from [GcsJavaClientImpl.java](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/fabf01d601537f9f663ff700e457809edbd172fe/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GcsJavaClientImpl.java) that was missing 2022 license (added in https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/939).

**Pull request merged**